### PR TITLE
Source Notion: add request size throttling on 504 errors

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What

Third attempt at resolving [oncall #3268](https://github.com/airbytehq/oncall/issues/3268). Customer is experiencing recurring sync failures due to 504 Gateway Timeout issues when syncing large datasets.

## How

- When a 504 error is encountered, the request page_size is reduced by half (to a minimum of 10) for each retried request to reduce burden on the server.
- Once a successful response has been received and the sync can proceed, page_size is reset to the default of 100 in parse_response.

## User Impact
Not a breaking change, only affects handling of retryable 504 errors

### CODE REVIEW

**Review by davydov-d — APPROVED** [review]
looks good!

File: airbyte-integrations/connectors/source-notion/source_notion/streams.py [hunk 1]
```diff
@@ -79,19 +79,30 @@ def check_invalid_start_cursor(response: requests.Response):
...
+        In the case of a 504 Gateway Timeout error, we can lower the page_size when retrying to reduce the load on the server.
         Docs: https://developers.notion.com/reference/errors#rate-limiting
-
         """
         retry_after = response.headers.get("retry-after", "5")
         if response.status_code == 429:
             return float(retry_after)
         if self.check_invalid_start_cursor(response):
             return 10
+        if response.status_code == 504:
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-notion/source_notion/streams.py:103]: why is it here?
seems like this should be added in should_retry

File: airbyte-integrations/connectors/source-notion/source_notion/streams.py [hunk 2]
```diff
@@ -121,6 +132,12 @@ def next_page_token(
             return {"next_cursor": next_cursor}
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+
+        # If page_size has been reduced after encountering a 504 Gateway Timeout error,
+        # we increase it back to the default of 100 once a success response is achieved, for the following API calls.
+        if response.status_code == 200 and self.page_size != 100:
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-notion/source_notion/streams.py:138]: same thing, please move to should_rerty to align with SRP principle

**Review by artem1205 — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-notion/source_notion/streams.py (hunk 2 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-notion/source_notion/streams.py]: Done and done! All logic to reduce/reset page_size has been moved to should_retry method

**Review by artem1205 — APPROVED** [review]

**Pull request merged**